src/core/information_functions.jl

```diff
@@ -281,12 +281,17 @@ function information(::InformationMeasure, ::DifferentialInfoEstimator, args...)
 end
 
 """
-    self_information(measure::InformationMeasure, pᵢ)
+    self_information(measure::InformationMeasure, p, N::Int)
 
-Compute the "self-information"/"surprisal" of a single probability `pᵢ` under the given 
-information measure. 
+Compute the "self-information"/"surprisal" of a single probability `p` under the given 
+information measure, assuming that `p` is part of a length-`N` probability distribution.
```

Comment:
Unfortunately I don't agree with the latest change of requiring `N`. It seems simpler, and more reasonable, to simply not allow Curado to be part of this interface. The opposite, defining the information unit as depending on N, doesn't make much sense at least not with how Shannon introduced it.

Reply:
@Datseris `Curado` is not the only information measure whose surprisal/self-information depends explicitly on `N`, when following the definition of an information measure as a probability weighted average of the surprisal (as I do in the paper).

> The opposite, defining the information unit as depending on N, doesn't make much sense at least not with how Shannon introduced it.

In the context of Shannon information unit alone, I agree. But the point of this interface is to _generalize_ the Shannon information unit. This inevitably introduces `N` as a parameter. 

Can we discuss the final API when I'm done with writing up the paper?  I'm not too far from finishing it; I just need to generate a few example applications. Since I am using this PR for the paper analyses,  it would be nice to not change anything in this draft PR until the paper is ready.

Reply:
The alternative is to have `information_content`/`information_unit` which dispatches to a subset of `InformationMeasure`s, and then `generalized_information_content`/`generalized_information_unit` which dispatches to those `InformationMeasure`s whose generalization of information unit depends on `N`. But that kind of defeats the purpose of having an interface to begin with - since we're back at defining multiple functions with different names for things that are fundamentally identical (modulo the parameter `N`).

Reply:
PS: I sent you a link to the paper draft, @Datseris